enhancements/network/optional-vips-with-extlb.md

```diff
@@ -0,0 +1,204 @@
+
+I don't believe this will impact MicroShift either, but even if it did it would
+reduce resources requirements slightly.
+
+### Implementation Details/Notes/Constraints
+
+A configuration to use this new functionality would look like the following:
+
+```yaml
+platform:
+  baremetal:
+    loadBalancer:
+      type: UserManaged
+```
+
+Note the lack of the apiVIPs and ingressVIPs fields, which previously would
+have been mandatory with this configuration.
+
+### Risks and Mitigations
+
+Not having internal DNS records for mandatory names slightly increases the
+chances of misconfiguration, but as most of these records are already
+required in the external DNS infrastructure it's not particularly concerning.
+
+### Drawbacks
+
+It will add complexity to the coredns configuration for on-prem IPI, but less
+than any other option we've come up with. It is also yet another possible
+variation in the networking configuration for on-prem deployments. That matrix
+is already impossibly large so it won't move the needle noticeably though.
+
+## Alternatives (Not Implemented)
+
+Two other options were considered, but rejected for the following reasons:
+
+* Support multiple VIPs in the coredns configuration. This would have
+  significantly increased the complexity of our coredns configuration
+  and would have introduced the possibility of variance between how DNS
+  resolves internally or externally. Additionally, multiple backends
+  behind a single DNS name is already supported by the external infrastructure
+  so at best we'd be duplicating existing functionality.
+
+* Remove coredns completely. This has a few problems. As noted earlier, it
+  would require changing the resolv-prepender service to handle the absence
+  of a local DNS server, and that's a much riskier change. It would also
+  eliminate name-based resolution between nodes.
+
+## Open Questions [optional]
+
+How will the validation for these fields work? Currently the VIP fields are
+always required, but with this feature they will only be mandatory if the
+external loadbalancer option is not set. I don't have enough experience with
+the API to know how that will work. Worst-case, we can add a validation to
+the installer though.
+
+How does this interact with the existing VSphere UPI configuration? The VIPs
+are already optional on that platform (which is how UPI is deployed), so we
+may need to rework some logic around that to account for the loadbalancer
+parameter as well. Alternatively we can just not support this on VSphere,
+but that may be undesirable.
+
+## Test Plan
+
+We will need to add the ability to configure api-int on the external DNS in
+our external loadbalancer tests. Since we already have an api record, that
+shouldn't be difficult.
+
+## Graduation Criteria
+
+This is a variation on an existing supported deployment scenario, so we don't
+plan to pursue a graduation process. It will be GA from initial implementation.
```

Comment:
Why should this new variation not go through the proper promotion channels before being supported?

Reply:
Honestly, we've had pretty bad luck with tech preview features for baremetal customers. They won't even try them.

However, given that I think this is going to be a more substantial change than I initially expected I'm discussing this with the field team asking for it.